Invalid type declarations for covariant field in AstroIntegration

The type declared for the Astro plugin specifies `unknown[]` for Vite's plugin config. This is incompatible with the real `AstroIntegration` type from astro.

```log
astro.config.mts:110:5 - error TS2322: Type 'AstroIntegration' is not assignable to type 'false | AstroIntegration | (false | AstroIntegration | null | undefined)[] | null | undefined'.
  Type 'AstroIntegration' is not assignable to type 'import("/home/lotus/Workspace/blog/node_modules/.pnpm/astro@4.0.9_patch_hash=76zgkjwwmhzrg4rf6mhio72waq_typescript@5.3.3/node_modules/astro/dist/@types/astro").AstroIntegration'.
    The types of 'hooks['astro:config:setup']' are incompatible between these types.
      Type '((options: { injectScript: AstroInjector; updateConfig: (config: { vite: { plugins: unknown[]; }; }) => void; }) => void | Promise<void>) | undefined' is not assignable to type '((options: { config: AstroConfig; command: "build" | "dev" | "preview"; isRestart: boolean; updateConfig: (newConfig: DeepPartial<AstroConfig>) => AstroConfig; ... 8 more ...; logger: AstroIntegrationLogger; }) => void | Promise<...>) | undefined'.
        Type '(options: { injectScript: AstroInjector; updateConfig: (config: { vite: { plugins: unknown[]; }; }) => void; }) => void | Promise<void>' is not assignable to type '(options: { config: AstroConfig; command: "build" | "dev" | "preview"; isRestart: boolean; updateConfig: (newConfig: DeepPartial<AstroConfig>) => AstroConfig; ... 8 more ...; logger: AstroIntegrationLogger; }) => void | Promise<...>'.
          Types of parameters 'options' and 'options' are incompatible.
            Type '{ config: AstroConfig; command: "build" | "dev" | "preview"; isRestart: boolean; updateConfig: (newConfig: DeepPartial<AstroConfig>) => AstroConfig; ... 8 more ...; logger: AstroIntegrationLogger; }' is not assignable to type '{ injectScript: AstroInjector; updateConfig: (config: { vite: { plugins: unknown[]; }; }) => void; }'.
              Types of property 'updateConfig' are incompatible.
                Type '(newConfig: DeepPartial<AstroConfig>) => AstroConfig' is not assignable to type '(config: { vite: { plugins: unknown[]; }; }) => void'.
                  Types of parameters 'newConfig' and 'config' are incompatible.
                    Type '{ vite: { plugins: unknown[]; }; }' is not assignable to type 'DeepPartial<AstroConfig>'.
                      The types of 'vite.plugins' are incompatible between these types.
                        Type 'unknown[]' is not assignable to type 'DeepPartial<PluginOption[] | undefined>'.
                          Type 'unknown' is not assignable to type 'PluginOption'.

110     imgit({}),
```

<details>
<summary>Print of the error formatted</summary>

![image](https://github.com/elringus/imgit/assets/11063910/32bed9b2-9ae2-4eb6-b27b-331121a4d9a4)
</details>

https://github.com/elringus/imgit/blob/e449e6fd12ee89884a23d950fcfd10ef51061e71/src/plugin/astro.ts#L11

That `plugins: unknown[]` is inside a function parameter, making the type of `updateConfig` contravariant over it. But the `updateConfig` field is itself part of a function parameter, meaning that `astro:config:setup` and by extension `AstroIntegration` contravariant over `updateConfig`. Transitively, `AstroIntegration` is _covariant_ over the `plugins` field.

This means that assigning this `AstroIntegration` type over Astro's actual type would require `unknown` to be assignable to Vite's `PluginOption`, but of course `unknown` is not assignable to anything except itself by definition (since it is a top type).

The field should either:
- Be properly typed using Vite's `PluginOption`
- Be typed as a subtype of Vite's `PluginOptions`, possibly with only the used properties
- Be typed with `any`, which is both a top and bottom type and accepts any assignment in both directions

--- a/src/plugin/astro.ts
+++ b/src/plugin/astro.ts
@@ -1,4 +1,4 @@
-﻿import { Prefs, Platform, Plugin, std } from "../server/index.js";
+import { Prefs, Platform, Plugin, std } from "../server/index.js";
 import { bind } from "../server/platform/index.js";
 import vite from "../plugin/vite.js";
 
@@ -8,7 +8,7 @@ declare type AstroIntegration = {
     hooks: {
         "astro:config:setup"?: (options: {
             injectScript: AstroInjector;
-            updateConfig: (config: { vite: { plugins: unknown[] } }) => void;
+            updateConfig: (config: { vite: { plugins: [ReturnType<typeof vite>] } }) => void;
         }) => void | Promise<void>;
     };
 };

